### PR TITLE
Add first "real" Redux sample code

### PR DESCRIPTION
**Pull request by brettdh**
## Description
I've added Redux placeholder start for auth and Patreon. In both cases,
it's a simple boolean - `state.auth.authorized` and `state.patreon.enabled` - 
but the main purpose is to show now we can structure state and organize
reducers. I'll make an effort to document this formally soon, but for now
this can at least be an example. For Patreon, the `enabled`/`disabled` is
just intended as a simulation of connecting Patreon or not; obviously
it needs to be replaced with real Patreon login and check Liturgists
subscription level.

Note that this doesn't yet include anything to do with asynchronous actions
such as fetching data from an API. That's my next PR, probably coming tomorrow. 😄 

## Motivation and Context
This is only intended as a sample for how to organize Redux state, actions,
selectors, and reducers, and as a starting point for discussion on those matters.

## How Has This Been Tested?
I navigated all throughout the screens in the app and didn't encounter any problems.
When Patreon is enabled, the Podcasts and Meditations screens have different text
compared to when Patreon is disabled.

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/navigation/MenuButton.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Text } from 'react-native';
+
+const MenuButton = ({ navigation }) => (
+  <Text onPress={() => navigation.navigate('DrawerOpen')}>
```
Comment (wittjosiah) [inline src/navigation/MenuButton.js:6]: Is `'DrawerOpen'` a constant that react-navigation is expecting?

File: src/styles/index.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import { StyleSheet } from 'react-native';
+
+export default StyleSheet.create({
```
Comment (wittjosiah) [inline src/styles/index.js:3]: I'm not sure its a good idea to separate styles from the components. These are fairly basic, foundation styles so its probably okay, but in general I think its better to keep the styles with the component they are styling. This allows you to understand what a component will look like without having to jump back and forth between the styles and also prevents you from reusing the styles without thinking more carefully about the overlap between the components.

**Review by wittjosiah — APPROVED** [review]

**Comment by brettdh** [comment]
> I'm not sure its a good idea to separate styles from the components.

I agree in general. I just don’t yet have any examples of styles that are
tightly coupled to their components. I suspect those will come up soon,
though, when we start picking up the tasks in #12.

Also, we probably want a generic `Screen` component that contains the
styles common to all screens - and that’s probably where these `container`
styles actually belong.

File: src/navigation/MenuButton.js (hunk 1 above)
Comment (brettdh) [reply src/navigation/MenuButton.js:6]: That's how I understand it, but I'm just going from one or two examples myself.

**Pull request merged**